pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -147,37 +142,16 @@ var (
 // to have a central consistent endpoint to get whether CA functionality is
 // enabled in istiod. EnableCA() is called in multiple places.
 func (s *Server) EnableCA() bool {
-	if !features.EnableCAServer {
-		return false
-	}
-	// Log if we're using self-signed certs without K8S, in debug mode
-	if s.kubeClient == nil {
```

Comment:
Why is this segment of code removed?

Reply:
This is already covered by `func (s *Server) createIstioCA(client corev1.CoreV1Interface, opts *caOptions) (*ca.IstioCA, error) {
`

Reply:
I did not find this segement of code in createIstioCA().
		// Without K8S the user needs to have the private key in a local file.
		// If that is missing - we'll generate an in-memory root for testing, and warn.
		signingKeyFile := path.Join(LocalCertDir.Get(), "ca-key.pem")
		if _, err := os.Stat(signingKeyFile); err != nil {
			log.Warnf("Will use in-memory root CA, no K8S access and no ca key file %s", signingKeyFile)
		}

Reply:
I mean this is duplicate with https://github.com/istio/istio/pull/30171/files/a611ae1cf6c2b87ff445a1d0a4c8300979441316#diff-0e039728d940bd528569ae3684ba2c8828df0f5d77456e3394fb0c94ea6e72e2L363